### PR TITLE
Chatty V3 hook

### PR DESCRIPTION
**Pull request by sub-kek**
Duplicate of https://github.com/DiscordSRV/DiscordSRV/pull/1648

But don't breaks compatibility with older chatty versions.

https://github.com/DiscordSRV/DiscordSRV/pull/1648#issuecomment-2229622133

### CODE REVIEW

File: src/main/java/github/scarsz/discordsrv/hooks/chat/ChattyV3ChatHook.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        try {
+            Class.forName("ru.brikster.chatty.api.ChattyApi");
+        } catch (ClassNotFoundException ignore) {
+            return false;
+        }
+        return true;
+    }
+
+}
```
Comment (PadowYT2) [inline src/main/java/github/scarsz/discordsrv/hooks/chat/ChattyV3ChatHook.java:96]:
```suggestion
}

```

File: settings.gradle.kts [hunk 2]
```diff
@@ -1 +1 @@
-rootProject.name = "DiscordSRV"
+rootProject.name = "DiscordSRV"
```
Comment (PadowYT2) [inline settings.gradle.kts:1]:
```suggestion
rootProject.name = "DiscordSRV"

```

**Comment by BANanaD3V** [comment]
Can you provide instructions on how to build, or, more specifically, add chatty v3 maven into mavenLocal?

**Comment by sub-kek** [comment]
> Can you provide instructions on how to build, or, more specifically, add chatty v3 maven into mavenLocal?

git clone https://github.com/Brikster/Chatty.git
cd Chatty
./gradlew publishToMavenLocal

Then build DiscordSRV with this PR.

**Comment by BANanaD3V** [comment]
It says my gradle version (9.0) is too high, what gradle is used here? 8?

**Comment by PadowYT2** [comment]
> It says my gradle version (9.0) is too high, what gradle is used here? 8?

Use `./gradlew` not `gradle`

**Comment by BANanaD3V** [comment]
> > It says my gradle version (9.0) is too high, what gradle is used here? 8?
> 
> Use `./gradlew` not `gradle`

I am

**Comment by BANanaD3V** [comment]
gradle-wrapper.properties say that it should be 8.5, but gradlew is somehow 9.0, strange

**Comment by PadowYT2** [comment]
> > > It says my gradle version (9.0) is too high, what gradle is used here? 8?
> > 
> > 
> > Use `./gradlew` not `gradle`
> 
> I am

You are better going to https://discord.discordsrv.com -> #coding and not here

**Comment by BANanaD3V** [comment]
Managed to build it using `gradle_8` package from nixpkgs, but it's strange that the gradlew is 9.0.

**Comment by sub-kek** [comment]
> Could not find ru.brikster:chatty-api:3.0.0-SNAPSHOT.

Publish it first...

**Review by granny — APPROVED** [review]
I've gone ahead and published Chatty v3 to scarsz' nexus repo. Everything LGTM

**Comment by Dinty1** [comment]
This still requires testing by a member of the team and for the latest commits to be pulled from develop so that the build works

**Review by Dinty1 — APPROVED** [review]
Tested and seems to work fine

**Comment by Brikster** [comment]
I confirm that there’re no plans to change Chatty v3 API after release :)

**Review by Vankka — APPROVED** [review]

**Pull request merged**